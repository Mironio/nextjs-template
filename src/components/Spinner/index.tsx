import spinner from '/public/images/spinner.svg';
import Image from 'next/image';
import en from '@/locales/en';

export default function Spinner({ classNames = '' }) {
  return (
    <div className={`${classNames} my-6 mt-24 flex items-center justify-center`}>
      <div className="rounded-full shadow-pq-m">
        <Image
          priority
          src={spinner as string}
          alt={en.loading}
          className="animate-spin"
        />
      </div>
    </div>
  );
}
