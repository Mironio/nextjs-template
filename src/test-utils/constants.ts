export const TestClassNames = 'vitest-1';
export const TestChildren = 'children';
