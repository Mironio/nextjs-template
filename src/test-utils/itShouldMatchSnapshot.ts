import { expect, it } from 'vitest';
import { JSX } from 'react';

export default function itShouldMatchSnapshot(jsx: JSX.Element) {
  it('should match snapshot', () => {
    expect(jsx).toMatchSnapshot();
  });
}
