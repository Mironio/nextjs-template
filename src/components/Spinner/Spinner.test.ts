import itShouldMatchSnapshot from '@/test-utils/itShouldMatchSnapshot';
import { TestClassNames } from '@/test-utils/constants';
import Spinner from './index';

itShouldMatchSnapshot(Spinner({ classNames: TestClassNames }));
