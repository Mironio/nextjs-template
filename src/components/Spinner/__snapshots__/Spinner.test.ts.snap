// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<div
  className="vitest-1 my-6 mt-24 flex items-center justify-center"
>
  <div
    className="rounded-full shadow-pq-m"
  >
    <ForwardRef
      alt="Loading"
      className="animate-spin"
      priority={true}
      src="/public/images/spinner.svg"
    />
  </div>
</div>
`;
